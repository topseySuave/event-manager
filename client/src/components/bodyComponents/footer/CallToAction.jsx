import React, { Component } from 'react';
import { Link } from 'react-router-dom';

/**
 * CallToAction Class Component
 * */
class CallToAction extends Component {
  /**
   * componentDidMount Method
   * @returns { void }
   * */
  componentDidMount() {
    $('.parallax').parallax();
  }

  /**
   * render Method
   * @returns { Component }
   * */
  render() {
    return (
      <div className="parallax-container">
        <div className="parallax">
          <img src="image/5.jpg" alt="parallax-img" />
        </div>
        <div className="container">
          <div className="center-align">
            <div className="row">
              <div className="col s12">
                <h4>Do you want to Manage your Own Event?</h4>
                <h5>View All Centers</h5>
                <br />
                <Link
                  to="centers"
                  className="btn waves-effect red darken-1"
                  href="centers"
                >
                  Centers
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default CallToAction;
