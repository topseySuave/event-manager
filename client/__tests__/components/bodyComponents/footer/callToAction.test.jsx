import React from 'react';
import { shallow } from 'enzyme';
import CallToAction from
  '../../../../src/components/bodyComponents/footer/CallToAction';

describe('CallToAction component', () => {
  const wrapper = shallow(<CallToAction />);
  test('should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
