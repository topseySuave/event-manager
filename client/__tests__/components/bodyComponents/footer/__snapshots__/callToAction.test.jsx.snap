// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction component CallToAction component should mount 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CallToAction />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="parallax"
        >
          <img
            alt="parallax-img"
            src="image/5.jpg"
          />
        </div>,
        <div
          className="container"
        >
          <div
            className="center-align"
          >
            <div
              className="row"
            >
              <div
                className="col s12"
              >
                <h4>
                  Do you want to Manage your Own Event?
                </h4>
                <h5>
                  View All Centers
                </h5>
                <br />
                <Link
                  className="btn waves-effect red darken-1"
                  href="centers"
                  replace={false}
                  to="centers"
                >
                  Centers
                </Link>
              </div>
            </div>
          </div>
        </div>,
      ],
      "className": "parallax-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="parallax-img"
            src="image/5.jpg"
          />,
          "className": "parallax",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "parallax-img",
            "src": "image/5.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="center-align"
          >
            <div
              className="row"
            >
              <div
                className="col s12"
              >
                <h4>
                  Do you want to Manage your Own Event?
                </h4>
                <h5>
                  View All Centers
                </h5>
                <br />
                <Link
                  className="btn waves-effect red darken-1"
                  href="centers"
                  replace={false}
                  to="centers"
                >
                  Centers
                </Link>
              </div>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col s12"
              >
                <h4>
                  Do you want to Manage your Own Event?
                </h4>
                <h5>
                  View All Centers
                </h5>
                <br />
                <Link
                  className="btn waves-effect red darken-1"
                  href="centers"
                  replace={false}
                  to="centers"
                >
                  Centers
                </Link>
              </div>
            </div>,
            "className": "center-align",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col s12"
              >
                <h4>
                  Do you want to Manage your Own Event?
                </h4>
                <h5>
                  View All Centers
                </h5>
                <br />
                <Link
                  className="btn waves-effect red darken-1"
                  href="centers"
                  replace={false}
                  to="centers"
                >
                  Centers
                </Link>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Do you want to Manage your Own Event?
                  </h4>,
                  <h5>
                    View All Centers
                  </h5>,
                  <br />,
                  <Link
                    className="btn waves-effect red darken-1"
                    href="centers"
                    replace={false}
                    to="centers"
                  >
                    Centers
                  </Link>,
                ],
                "className": "col s12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Do you want to Manage your Own Event?",
                  },
                  "ref": null,
                  "rendered": "Do you want to Manage your Own Event?",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "View All Centers",
                  },
                  "ref": null,
                  "rendered": "View All Centers",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Centers",
                    "className": "btn waves-effect red darken-1",
                    "href": "centers",
                    "replace": false,
                    "to": "centers",
                  },
                  "ref": null,
                  "rendered": "Centers",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="parallax"
          >
            <img
              alt="parallax-img"
              src="image/5.jpg"
            />
          </div>,
          <div
            className="container"
          >
            <div
              className="center-align"
            >
              <div
                className="row"
              >
                <div
                  className="col s12"
                >
                  <h4>
                    Do you want to Manage your Own Event?
                  </h4>
                  <h5>
                    View All Centers
                  </h5>
                  <br />
                  <Link
                    className="btn waves-effect red darken-1"
                    href="centers"
                    replace={false}
                    to="centers"
                  >
                    Centers
                  </Link>
                </div>
              </div>
            </div>
          </div>,
        ],
        "className": "parallax-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="parallax-img"
              src="image/5.jpg"
            />,
            "className": "parallax",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "parallax-img",
              "src": "image/5.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="center-align"
            >
              <div
                className="row"
              >
                <div
                  className="col s12"
                >
                  <h4>
                    Do you want to Manage your Own Event?
                  </h4>
                  <h5>
                    View All Centers
                  </h5>
                  <br />
                  <Link
                    className="btn waves-effect red darken-1"
                    href="centers"
                    replace={false}
                    to="centers"
                  >
                    Centers
                  </Link>
                </div>
              </div>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
              >
                <div
                  className="col s12"
                >
                  <h4>
                    Do you want to Manage your Own Event?
                  </h4>
                  <h5>
                    View All Centers
                  </h5>
                  <br />
                  <Link
                    className="btn waves-effect red darken-1"
                    href="centers"
                    replace={false}
                    to="centers"
                  >
                    Centers
                  </Link>
                </div>
              </div>,
              "className": "center-align",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col s12"
                >
                  <h4>
                    Do you want to Manage your Own Event?
                  </h4>
                  <h5>
                    View All Centers
                  </h5>
                  <br />
                  <Link
                    className="btn waves-effect red darken-1"
                    href="centers"
                    replace={false}
                    to="centers"
                  >
                    Centers
                  </Link>
                </div>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4>
                      Do you want to Manage your Own Event?
                    </h4>,
                    <h5>
                      View All Centers
                    </h5>,
                    <br />,
                    <Link
                      className="btn waves-effect red darken-1"
                      href="centers"
                      replace={false}
                      to="centers"
                    >
                      Centers
                    </Link>,
                  ],
                  "className": "col s12",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Do you want to Manage your Own Event?",
                    },
                    "ref": null,
                    "rendered": "Do you want to Manage your Own Event?",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "View All Centers",
                    },
                    "ref": null,
                    "rendered": "View All Centers",
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Centers",
                      "className": "btn waves-effect red darken-1",
                      "href": "centers",
                      "replace": false,
                      "to": "centers",
                    },
                    "ref": null,
                    "rendered": "Centers",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
